docs/keto/modeling/learn-the-concepts.mdx

```diff
@@ -0,0 +1,360 @@
+
+Notice that our permission model is hierarchical, in the sense that everybody that can `view` can also `edit`, and everybody that
+can `edit` can also `delete` and `share`.
+
+Therefore, we can write the `view` permission like this:
+
+```diff
+  view: (ctx: Context): boolean =>
+    this.related.viewers.includes(ctx.subject) ||
+-   this.related.editors.includes(ctx.subject) ||
+-   this.related.owners.includes(ctx.subject) ||
+-   this.related.parents.traverse((parent) => parent.permits.view(ctx))
++   this.permits.edit(ctx)
+```
+
+Applying this to the whole config yields:
+
+```ts title="permissions-v5.ts"
+import { Namespace, Context } from "@ory/keto-namespace-types"
+
+class User implements Namespace {}
+
+class Document implements Namespace {
+  related: {
+    owners: User[]
+    editors: User[]
+    viewers: User[]
+    parents: Folder[]
+  }
+
+  permits = {
+    view: (ctx: Context): boolean =>
+      this.related.viewers.includes(ctx.subject) ||
+      //highlight-next-line
+      this.permits.edit(ctx),
+    edit: (ctx: Context): boolean =>
+      this.related.editors.includes(ctx.subject) ||
+      //highlight-next-line
+      this.permits.share(ctx),
+    delete: (ctx: Context): boolean =>
+      //highlight-next-line
+      this.permits.share(ctx),
+    share: (ctx: Context): boolean =>
+      this.related.owners.includes(ctx.subject) || this.related.parents.traverse((parent) => parent.permits.share(ctx)),
+  }
+}
+
+class Folder implements Namespace {
+  related: {
+    owners: User[]
+    editors: User[]
+    viewers: User[]
+    parents: Folder[]
+  }
+
+  permits = {
+    delete: (ctx: Context): boolean =>
+      //highlight-next-line
+      this.permits.share(ctx),
+    share: (ctx: Context): boolean =>
+      this.related.owners.includes(ctx.subject) || this.related.parents.traverse((parent) => parent.permits.share(ctx)),
+  }
+}
+```
+
+Whether or not this refatoring makes sense in your application depends on your requirements, of course. The take-away is that
+these kind of refacorings are possible because the permission config is essentially just code. So you can apply the same techiques
+your permission config as to your code to create well-structured and maintainable software systems.
+
+### Test permissions
+
+It is important to test your permission model. To explore the permissions, you can
+[create relationships and check permissions](../guides/simple-access-check-guide.mdx) through the API or SDK.
+
+For continuous testing, we recommend the following best practices:
+
+- Automate testing your permission model. Write a test, in a language of your choice, that inserts the relationships and checks
+  the permissions through the [SDK](../sdk/01_overview.md).
+- For complex permission model changes, use a separate Ory Network project. Each Ory Network project has an isolated permission
+  model, so you can iterate on and test your changes on a test project and only deploy the changes once all tests pass.
```

Comment:
> Each Ory Network project has an isolated permission model

This document talks about CREATING a permission model. Does every project have one by default? Or did you mean that every project is an isolated env. that can have its own permission model?

Reply:
I meant that every that each project has its own (isolated) permission model.